pkg/controller/upload-controller.go

```diff
@@ -500,7 +544,19 @@ func getScratchPvcName(name string) string {
 }
 
 // getUploadResourceName returns the name given to upload resources
-func getUploadResourceName(name string) string {
+func getUploadResourceNameFromPvc(pvc *corev1.PersistentVolumeClaim) string {
+	podName, ok := pvc.Annotations[AnnUploadPod]
+	if ok {
+		return podName
+	}
+
+	// fallback to legacy naming, in fact the following function is fully compatible with legacy
+	// name concatenation "cdi-upload-{pvc.Name}" if the name length is under the size limits,
+	return naming.GetResourceName("cdi-upload", pvc.Name)
+}
+
+// createUploadResourceName returns the name given to upload resources
+func createUploadResourceName(name string) string {
```

Comment:
So I am a little confused at this point, this function is called always with the pvc name, and the result is stored in the annotation. But why are we doing that? This should be a deterministic result based on the pvc name, so instead of storing and reading from the annotation, why not simply call this function whenever we are interested in finding the pod name based on the pvc name?

Reply:
So you are right, our function is now deterministic, and currently storing in annotation does not help us. So it has no value now, but might help us in the future:
- if we decide to resolve duplicate names/conflicts by trying some other name. 
- if we decide to make name generation more random/unique. 
- if we decide to 'change' the naming method, we might use annotation for backwards compatibility.

Now after you comment, I think, this "store in annotation" commit", and the next one, are nice to have, but really are maybe over engineering for current state. 

I mean - keeping it here, and improving some tests might be beneficial (for the future needs?), but removing it - might simplify the code. So I am open for hearing arguments for one of this directions. Use annotations to store names, or do not (since naming is deterministic)
 
@mhenriks @awels What do you think?